src/master/production/sync-replication.md

```diff
@@ -0,0 +1,199 @@
+
+The primary server keeps a record of the current state of each replica. The current state of each replica is updated through the `ExportTx` grpc call from the replica server. So when a new transaction request comes to the primary server, it precommits the transaction, and checks if a quorum (on the transaction) has been reached by the replica server(s) by checking their state continously. If the quorum was reached, the transaction is marked as successful.
+
+<div class="wrapped-picture">
+
+![how synchronous replication works](/immudb/replication-state.png)
+
+</div>
+
+</WrappedSection>
+
+<WrappedSection>
+
+### Deciding on number of servers in a cluster
+
+Synchronous replication in a cluster can function only if the majority of servers are up and running. In systems doing data replication, it is important to consider the throughput of write operations. Every time data is written to the cluster, it needs to be copied to multiple replicas. Every additional server adds some overhead to complete this write. The latency of data write is directly proportional to the number of servers forming the quorum.
+
+</WrappedSection>
+
+<WrappedSection>
+### Settings
+
+Synchronous replication is enabled per database. The following flags in the `immuadmin` tool will help in setting up synchronous replication for your database.
+
+```bash
+Flags:
+      --replication-allow-tx-discarding              allow precommitted transactions to be discarded if the follower diverges from the master
+      --replication-commit-concurrency uint32        set the number of threads for concurrent replication
+      --replication-follower-password string         set password used for replication
+      --replication-follower-username string         set username used for replication
+      --replication-is-replica                       set database as a replica
+      --replication-master-address string            set master address
+      --replication-master-database string           set master database to be replicated
+      --replication-master-port uint32               set master port
+      --replication-prefetch-tx-buffer-size uint32   maximum number of prefeched transactions (default 100)
+      --replication-sync-enabled                     enable synchronous replication
+      --replication-sync-followers uint32            set a minimum number of followers for ack replication before transactions can be committed
+```
+
+</WrappedSection>
+
+<WrappedSection>
+
+### Setup
+
+This setup guides you through a simple demonstration of how synchronous replication works in ImmuDB. Starting with a 2-node local cluster, you'll write some data and verify that it replicates in sync.
+#### Before you begin
+
+Make sure you already have [ImmuDB installed]((../running/download.md)).
+
+    Since you're running a local cluster, all nodes use the same hostname (`localhost`).
```

Comment:
Looks like this renders incorrectly:

![image](https://user-images.githubusercontent.com/1365626/194288246-7b0be57b-0bab-4682-b372-941c923f5fce.png)

Maybe better to use quotation?

```suggestion
> Since you're running a local cluster, all nodes use the same hostname (`localhost`).
```

Reply:
added the quotation